Objective-C/CBLQueryResultSet.mm

```diff
@@ -110,17 +120,29 @@ - (id) currentObject {
 
 // Called by CBLQueryResultsArray
 - (id) objectAtIndex: (NSUInteger)index {
-    if (!c4queryenum_seek(_c4enum, index, &_error)) {
-        NSString* message = sliceResult2string(c4error_getMessage(_error));
-        [NSException raise: NSInternalInconsistencyException
-                    format: @"CBLQueryEnumerator couldn't get a value: %@", message];
+    CBLDatabase* strongDB = self.database;
+    Assert(strongDB, @"Database has been released.");
```

Comment:
If this check is going to happen every time self.database is called, maybe move it into the `database` method.